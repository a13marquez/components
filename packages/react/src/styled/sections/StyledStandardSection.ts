import styled from 'styled-components';
import { StandardStyles } from '../shared-styles';

export const StyledStandardSection = styled.section`
  ${StandardStyles}
`