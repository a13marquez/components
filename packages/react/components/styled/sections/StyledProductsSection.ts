import styled from 'styled-components';
import { StyledStandardSection } from './StyledStandardSection';


export const StyledProductsSection = styled(StyledStandardSection)`
  padding: 40px 32px 40px 56px;
  flex:1;
`