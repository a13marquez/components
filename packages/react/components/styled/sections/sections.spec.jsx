import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import { StyledStandardSection } from './StyledStandardSection';
import { StyledProductsSection } from './StyledProductsSection';

test('It renders Styled Standard Section', () => {
  const StandardSection = mount(<StyledStandardSection />);
  expect(StandardSection).toMatchSnapshot();
});

test('It renders Styled Product Section', () => {
  const ProductsSection = mount(<StyledProductsSection />);
  expect(ProductsSection).toMatchSnapshot();
});

test('Styled Standard Row have the correct styles', () => {
  const StandardSection = mount(<StyledStandardSection />);
  expect(StandardSection).toHaveStyleRule('margin', '0');
  expect(StandardSection).toHaveStyleRule('padding', '0');
  expect(StandardSection).toHaveStyleRule('border', '0');
  expect(StandardSection).toHaveStyleRule('vertical-align', 'baseline');
  expect(StandardSection).toHaveStyleRule('font', '100%');
});

test('Styled Product Title Row have the correct styles', () => {
  const ProductsSection = mount(<StyledProductsSection />);
  expect(ProductsSection).toHaveStyleRule('margin', '0');
  expect(ProductsSection).toHaveStyleRule('border', '0');
  expect(ProductsSection).toHaveStyleRule('vertical-align', 'baseline');
  expect(ProductsSection).toHaveStyleRule('font', '100%');

  expect(ProductsSection).toHaveStyleRule('padding', '40px 32px 40px 56px');
  expect(ProductsSection).toHaveStyleRule('flex', '1');
});
