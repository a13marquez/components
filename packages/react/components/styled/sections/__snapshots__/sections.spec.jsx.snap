// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders Styled Product Section 1`] = `ReactWrapper {}`;

exports[`It renders Styled Standard Section 1`] = `ReactWrapper {}`;
